--- vlc-android/libvlc/src/org/videolan/libvlc/VideoHelper.java.orig	2019-04-04 13:54:08.415591000 +0900
+++ vlc-android/libvlc/src/org/videolan/libvlc/VideoHelper.java	2019-04-05 14:06:52.229001000 +0900
@@ -194,13 +194,16 @@
         int sh;
 
         // get screen size
-        if (activity != null) {
-            sw = activity.getWindow().getDecorView().getWidth();
-            sh = activity.getWindow().getDecorView().getHeight();
-        } else if (mDisplayManager != null && mDisplayManager.getPresentation() != null && mDisplayManager.getPresentation().getWindow() != null) {
+        if (mDisplayManager != null && mDisplayManager.getPresentation() != null && mDisplayManager.getPresentation().getWindow() != null) {
             sw = mDisplayManager.getPresentation().getWindow().getDecorView().getWidth();
             sh = mDisplayManager.getPresentation().getWindow().getDecorView().getHeight();
-        } else return;
+        } else {
+			/**
+			*  DecorViewのサイズを元にせず、全画面表示以外もできるようにする。
+			* */
+            sw = mVideoSurfaceFrame.getWidth();
+            sh = mVideoSurfaceFrame.getHeight();
+        }
 
         // sanity check
         if (sw * sh == 0) {